core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -55,77 +56,132 @@ object ConsumerGroupCommand {
+    println("Error: " + msg)
+    e.map(debug("Exception in consumer group command", _))
+  }
+
+  def printMessage(msg: String): Unit = println(msg)
+
+  def printAssignment(groupAssignment: Seq[PartitionAssignmentState]): Unit = {
+    println("Consumer group: " + groupAssignment.head.group)
+    groupAssignment.head.coordinator match {
+      case Some(node) => println("Consumer group coordinator broker id: " + node.id)
+      case default =>
+    }
+    println("\n%-30s %-10s %-15s %-15s %-10s %-50s %-30s %s".format("TOPIC", "PARTITION", "CURRENT-OFFSET", "LOG-END-OFFSET", "LAG", "MEMBER-ID", "MEMBER-HOST", "CLIENT-ID"))
+    groupAssignment.foreach { memberAssignment =>
+      println("%-30s %-10s %-15s %-15s %-10s %-50s %-30s %s".format(
+        memberAssignment.topic.getOrElse(""), memberAssignment.partition.getOrElse(""),
+        memberAssignment.offset.getOrElse(""), memberAssignment.logEndOffset.getOrElse(""), memberAssignment.lag.getOrElse(""),
+        memberAssignment.memberId.getOrElse("none"), memberAssignment.memberHost.getOrElse(""), memberAssignment.clientId.getOrElse("")))
+    }
+  }
+
+  protected case class PartitionAssignmentState(group: String, coordinator: Option[Node], topic: Option[String],
+                                                partition: Option[Int], offset: Option[Long], lag: Option[Long],
+                                                memberId: Option[String], memberHost: Option[String],
+                                                clientId: Option[String], logEndOffset: Option[Long])
+
   sealed trait ConsumerGroupService {
 
-    def list(): Unit
+    def listGroups(): List[String]
 
-    def describe() {
-      describeGroup(opts.options.valueOf(opts.groupOpt))
+    def describeGroup(): (String, Option[Seq[PartitionAssignmentState]]) = {
+      collectGroupAssignment(opts.options.valueOf(opts.groupOpt))
     }
 
     def close(): Unit
 
     protected def opts: ConsumerGroupCommandOptions
 
-    protected def getLogEndOffset(topic: String, partition: Int): LogEndOffsetResult
-
-    protected def describeGroup(group: String): Unit
-
-    protected def describeTopicPartition(group: String,
-                                         topicPartitions: Seq[TopicAndPartition],
-                                         getPartitionOffset: TopicAndPartition => Option[Long],
-                                         getOwner: TopicAndPartition => Option[String]): Unit = {
-      topicPartitions
-        .sortBy { case topicPartition => topicPartition.partition }
-        .foreach { topicPartition =>
-          describePartition(group, topicPartition.topic, topicPartition.partition, getPartitionOffset(topicPartition),
-            getOwner(topicPartition))
-        }
+    protected def getLogEndOffset(topicPartition: TopicPartition): LogEndOffsetResult
+
+    protected def collectGroupAssignment(group: String): (String, Option[Seq[PartitionAssignmentState]])
```

Comment:
Since the zookeeper service has no notion of consumer state, should we return `Option[String]` instead?